README.md

```diff
@@ -91,97 +107,168 @@ cloning over SSH may provide a better experience (which requires
     git clone git@github.com:apple/swift.git
     ./swift/utils/update-checkout --clone-with-ssh
 
-#### CMake
-[CMake](https://cmake.org) is the core infrastructure used to configure builds of 
-Swift and its companion projects; at least version 3.4.3 is required. 
+### Building Swift
 
-On macOS, you can download the [CMake Binary Distribution](https://cmake.org/download),
-bundled as an application, copy it to `/Applications`, and add the embedded
-command line tools to your `PATH`:
+The `build-script` is a high-level build automation script that supports basic
+options such as building a Swift-compatible LLDB, building the Swift Package
+Manager, building for various platforms, running tests after builds, and more.
 
-    export PATH=/Applications/CMake.app/Contents/bin:$PATH
-    
-On Linux, if you have not already installed Swift's [development
-dependencies](#linux-dependencies), you can download and install the CMake
-package separately using the following command:
+There are two primary build systems to use: Xcode and Ninja. The Xcode build
+system allows you to work in Xcode.app, but Ninja is a bit faster and supports
+more environments.
 
-    sudo apt-get install cmake
-    
+To build using Ninja, run:
 
-#### Ninja
-[Ninja](https://ninja-build.org) is the current recommended build system
-for building Swift and is the default configuration generated by CMake. [Pre-built
-packages](https://github.com/ninja-build/ninja/wiki/Pre-built-Ninja-packages)
-are available for macOS and Linux distributions. You can also clone Ninja
-next to the other projects and it will be bootstrapped automatically:
+    utils/build-script --release-debuginfo
 
-**Via HTTPS**
+If you are interested in debugging the compiler, run:
```

Comment:
This isn't just the compiler, but the whole Swift project: Clang, Swift frontend, LLVM, standard library, runtimes, and all. I would move this section lower. 

Straw man: """
When working on a specific area of the project, it helps to build the area you're working on in a debug configuration while building the rest of the project with optimizations. Below are some examples of using debug variants:
```
utils/build-script --release-debuginfo --debug-swift # Swift frontend built in debug
utils/build-script --release-debuginfo --debug-swift-stdlib # Standard library built in debug
utils/build-script --release-debuginfo --debug-swift --force-optimized-typechecker # Swift frontend sans type checker built in debug
```

If you want to build the entire project in debug, which will produce a considerably slower compiler and standard library, you can run:
```
utils/build-script --debug
```
"""